File: tests/fixtures/openai-realtime/openai-realtime-migrate-beta-to-ga-broken/02-agent-socket.ts
import WebSocket from 'ws';

class AudioInterceptor {
  private agentOpenAISocket?: WebSocket;

  setup(apiKey: string) {
    const url = 'wss://api.openai.com/v1/realtime?model=gpt-realtime';
    this.agentOpenAISocket = new WebSocket(url, {
      headers: {
        Authorization: `Bearer ${apiKey}`,
        'OpenAI-Beta': 'realtime=v1',
      },
    });
  }
}

export default AudioInterceptor;

File: tests/fixtures/openai-realtime/openai-realtime-migrate-beta-to-ga-broken/02_agent_socket.py
import os

import websockets

AGENT_URL = "wss://api.openai.com/v1/realtime?model=gpt-realtime"


async def connect_agent_socket():
    agent_ws = await websockets.connect(
        AGENT_URL,
        additional_headers={
            "Authorization": f"Bearer {os.environ.get('OPENAI_API_KEY')}",
            "OpenAI-Beta": "realtime=v1",
        },
    )
    return agent_ws
